OS4 define and makefile updates

charabaruk opened the pull request.

charabaruk committed:
Use inline functions for AmigaOS 4.x

charabaruk committed:
Support obsolete ASL struct member names for OS 4.x

charabaruk committed:
Fix hook definitions for AOS 4.x

charabaruk committed:
Clean AOS4 build artifacts, simplify clean command

charabaruk committed:
Fix up VS Code settings for AmigaOS 4.x

Provide more appropriate paths for the 4.x SDK and for MUI, based on
default drawer layout for install on AmigaOS 4.x itself. MUI developer
files on 4.x are installed in a "MUI" drawer within the SDK installation
instead of in a "Developer" drawer within the MUI installation, unless
the user manually installs and configures MUI. SDK installs themselves
replace older versions of the 4.x SDK, so there's no version number in
the drawer name.

charabaruk committed:
Properly ignore AOS4 executable from git

walkero-gr commented on src/funcs.c:
```diff
@@ -49,6 +49,9 @@
 #include <exec/memory.h>
 #include <exec/types.h>
+#if defined(__amigaos4__)
+#define ASL_PRE_V38_NAMES
```
Can you please elaborate about this line? Why is this needed?

charabaruk replied:
With AmigaOS 4.0 and onward, ASL requester structures have had pretty much all their members renamed. However, the old names are still available when `ASL_PRE_V38_NAMES` is defined ahead of including libraries/asl.h. This takes care of another compilation problem I was having with your branch, without needing another define set in `$CFLAGS_AOS4` in the makefiles -- since this is the only place where libraries/asl.h is included, it made sense to me to only define `ASL_PRE_V38_NAMES` alongside it.

charabaruk committed:
Use same environment variables for Docker and regular vbcc builds

Require that $NDK_INC and $MUI_INC are defined rather than expect that NDK
and MUI header files are in fixed locations.

charabaruk committed:
Use separate env var for OS4 include paths

charabaruk committed:
Include AOS4 binary in release packages
